fabric-entity-events-v1/src/main/java/net/fabricmc/fabric/api/entity/event/v1/EntityElytraEvents.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.entity.event.v1;
+
+import net.minecraft.entity.EquipmentSlot;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.item.ItemStack;
+
+import net.fabricmc.fabric.api.event.Event;
+import net.fabricmc.fabric.api.event.EventFactory;
+
+/**
+ * Events related to elytra flight for living entities. Elytra flight is also knows as "fall flying".
+ */
+public final class EntityElytraEvents {
+	/**
+	 * An event to check if elytra flight (both through normal and custom elytras) is allowed.
+	 * All listeners need to return true to allow the entity to fly, otherwise elytra flight will be blocked/stopped.
+	 */
+	public static final Event<Allow> ALLOW = EventFactory.createArrayBacked(Allow.class, listeners -> livingEntity -> {
+		for (Allow listener : listeners) {
+			if (!listener.allowElytraFlight(livingEntity)) {
+				return false;
+			}
+		}
+
+		return true;
+	});
+
+	/**
+	 * An event to grant elytra flight to living entities when some condition is met.
+	 * Will be called when players try to start elytra flight by pressing space in mid-air, and every tick for all living entities to check if elytra flight is still allowed.
+	 *
+	 * <p>Items that wish to enable custom elytra flight when worn in the chest equipment slot can simply implement {@link FabricElytraItem} instead of registering a listener.
+	 */
+	public static final Event<Custom> CUSTOM = EventFactory.createArrayBacked(Custom.class, listeners -> (livingEntity, damageElytra) -> {
+		for (Custom listener : listeners) {
+			if (listener.useCustomElytra(livingEntity, damageElytra)) {
+				return true;
+			}
+		}
+
+		return false;
+	});
+
+	static {
+		CUSTOM.register((livingEntity, damageElytra) -> {
+			ItemStack chestStack = livingEntity.getEquippedStack(EquipmentSlot.CHEST);
+
+			if (chestStack.getItem() instanceof FabricElytraItem fabricElytraItem) {
+				return fabricElytraItem.useCustomElytra(livingEntity, chestStack, damageElytra);
+			}
+
+			return false;
+		});
+	}
+
+	@FunctionalInterface
+	public interface Allow {
+		/**
+		 * @return false to block elytra flight, true to allow it (unless another listener returns false)
+		 */
+		boolean allowElytraFlight(LivingEntity livingEntity);
```

Comment:
(`livingEntity` is unnecessarily verbose, but changing it for the sake of bikeshedding is pointless)